main.tf

```diff
@@ -14,10 +14,10 @@ locals {
   default_service_role_enabled        = local.enabled && var.create_default_service_role
   default_service_role_count          = local.default_service_role_enabled ? 1 : 0
   service_role_arn                    = local.default_service_role_enabled ? join("", aws_iam_role.default.*.arn) : var.service_role_arn
-  default_policy_name = {
-    Server = "AWSCodeDeployRole"
-    Lambda = "AWSCodeDeployRoleForLambda"
-    ECS    = "AWSCodeDeployRoleForECS"
+  default_policy_arn = {
+    Server = "arn:aws:iam::aws:policy/service-role/AWSCodeDeployRole"
+    Lambda = "arn:aws:iam::aws:policy/service-role/AWSCodeDeployRoleForLambda"
+    ECS    = "arn:aws:iam::aws:policy/AWSCodeDeployRoleForECS"
```

Comment:
better to use the current partition here instead of hard coding aws. this allows govcloud and China regions.

https://registry.terraform.io/providers/hashicorp/aws/latest/docs/data-sources/partition

Reply:
sure - will do this now. Thank you for looking at this so quickly! It's much appreciated

Reply:
```
diff --git a/main.tf b/main.tf
index 599df32..a86ea16 100644
--- a/main.tf
+++ b/main.tf
@@ -15,9 +15,9 @@ locals {
   default_service_role_count          = local.default_service_role_enabled ? 1 : 0
   service_role_arn                    = local.default_service_role_enabled ? join("", aws_iam_role.default.*.arn) : var.service_role_arn
   default_policy_arn = {
-    Server = "arn:aws:iam::aws:policy/service-role/AWSCodeDeployRole"
-    Lambda = "arn:aws:iam::aws:policy/service-role/AWSCodeDeployRoleForLambda"
-    ECS    = "arn:aws:iam::aws:policy/AWSCodeDeployRoleForECS"
+    Server = "arn:${data.aws_partition.current.partition}:iam::aws:policy/service-role/AWSCodeDeployRole"
+    Lambda = "arn:${data.aws_partition.current.partition}:iam::aws:policy/service-role/AWSCodeDeployRoleForLambda"
+    ECS    = "arn:${data.aws_partition.current.partition}:iam::aws:policy/AWSCodeDeployRoleForECS"
   }
 }

@@ -35,6 +35,8 @@ data "aws_iam_policy_document" "assume_role" {
   }
 }

+data "aws_partition" "current" {}
+
```

Changed as requested.